### ISSUE TITLE
Call-methylation ends with [E::sam_itr_next] Null iterator

### ISSUE THREAD

**Issue by raquelgarza** [jts/nanopolish#1007]
Hi, 

I'm trying to run call-methylation and facing the following error:

```
[bam process] iterating over region: chrX:71367000-71469000
[E::sam_itr_next] Null iterator
```

My command looks like:

`nanopolish call-methylation -t 4 -r file.fastq.gz -b file.sorted.bam -g genome.fa -w "chrX:71367000-71469000" > methylation_calls.tsv`

I also tried 

`nanopolish call-methylation -t 4 -r file.fastq.gz -b file.sorted.bam -g genome.fa -o methylation_calls.tsv "chrX:71367000-71469000"`

Both end with the same error message. 

Any suggestions?

Thanks!!

Note: I saw a previous issue in this github about the same error message but for eventalign, so I double checked that my bam file is not empty.

**Comment by raquelgarza** [comment]
Hi! Yes, it's in the same directory as the bam file.

**Comment by jts** [comment]
Thanks. Do you get an error message if you run:

```
samtools view file.sorted.bam chrX:71367000-71469000
```

**Comment by raquelgarza** [comment]
Yes... `[main_samview] region "chrX:71367000-71469000" specifies an invalid region or unknown reference. Continue anyway.`

This was very strange because I could have sworn this region was on my genome index. But I just realized that when I created the index for minimap2 it just took one of my fasta files for consideration. 

For anyone encountering this error, maybe check `samtools view -H file.bam` to make sure the region exists...

Thank you @jts, I'll close this issue now.
